app/commands/v2/patch_link_set.rb

```diff
@@ -25,13 +25,16 @@ def call
         # we need to reload the link_set as the links association will be stale
         link_set.reload
 
-        orphaned_content_ids = link_diff_between(links_before_patch, link_set.links.map(&:target_content_id))
+        orphaned_content_ids = link_diff_between(before_links.map(&:target_content_id), link_set.links.map(&:target_content_id))
 
         after_transaction_commit do
           send_downstream(orphaned_content_ids)
         end
 
-        Action.create_patch_link_set_action(link_set, event)
+        action = Action.create_patch_link_set_action(link_set, event)
+
+        after_links = link_set.links.to_a
+        LinkChangeService.new(action, before_links, after_links).record
```

Comment:
I feel like this LinkChange code belongs inside the `create_patch_link_set_action` method. Logic being that these aren't things that live independently and you wouldn't want to be creating patch_link_set actions without this data.

Reply:
Excellent point